### DISCUSSION TITLE
Noise GUI

### DISCUSSION THREAD

**Discussion by attackgoat** [Razaekel/noise-rs#340] [Show and tell]
I created a desktop/web editor using `egui` + `egui_snarl` + `eframe`:
![image](https://github.com/Razaekel/noise-rs/assets/7445438/8f97da3a-0efc-4151-8aaa-92d9e392e71b)
https://github.com/attackgoat/noise_gui
(link to [web version](https://attackgoat.github.io/noise_gui/))

All noise functions/parameters are supported and you can visually debug a noise graph without having to (_tediously_) save results to file.

> [!TIP]
> Image previews are generated using multi-threading on desktop but are currently limited to a single thread on the web. Large graphs are _much_ faster on the desktop version.

I plan on implementing load/save functionality for the "project" file as well as the overall noise function including named parameter support. The goal is to be able to design noise visually and then load it up at runtime and provide the seed or other constant values and be able to sample the result.

Contributions or suggestions/issues are welcome!

**Comment by jeffreyrosenbluth** [comment]
Great work! I think this will be really helpful.
